### PR TITLE
Allow for custom messages; closes #400

### LINKED ISSUE #400: Add ability to customize/override rule message text
It would be really nice if there was an easy way to customize (or override) the default rule text for each rule when you declare the rule. For example:
````
var rules = {
	'rules': {
		'number-leading-zero': [ 2, 'never' , 'blah blah blah'],
	}
};
````
**Output:** `blah blah blah (number-leading-zero)`

**Where as of the default (no override), would be the normal:**

````
var rules = {
	'rules': {
		'number-leading-zero': [ 2, 'never'],
	}
};
````
**Output:** `Unexpected leading zero (number-leading-zero)`

Is this something that can easily be done?

### PR DESCRIPTION
**Pull request by davidtheclark**
Easy custom messages: just add them as a secondary `message` option to any rule.

Anybody who feels like testing this out and trying to break it before we release --- that would be grand!

cc/ @bluetidepro, @kangax, @jeddy3

### CODE REVIEW

File: src/postcssPlugin.js [hunk 1]
```diff
@@ -60,17 +61,20 @@ export default postcss.plugin("stylelint", (options = {}) => {
 
         const rawRuleSettings = config.rules[ruleName]
         const ruleSettings = normalizeRuleSettings(rawRuleSettings, ruleName)
+        const primaryOption = ruleSettings[0]
```
Comment (jeddy3) [inline src/postcssPlugin.js]: Nice. Adding these variables names here makes the subsequent code much clearer.

**Comment by jeddy3** [comment]
Oh wow, I didn't think you'd get around to giving this a shot this weekend. Nice one!

**Comment by jeddy3** [comment]
> Anybody who feels like testing this out and trying to break it before we release

I think there might be a bug.

`{ "message" : "blah" }` seems to work great for rules without secondary options :) However, if the rule validates its secondary options it'll complain about `"message"`:

`Invalid option name "message" for rule "indentation"`

FYI, this doesn't happen for `"warn"`... is `"warn"` excluded from the validator?

**Comment by davidtheclark** [comment]
@jeddy3 : Good catch. Will fix.

**Comment by davidtheclark** [comment]
@jeddy3 : Please try again.

**Comment by jeddy3** [comment]
Sweet. Works now.

I'll wait for appveyor to finish chugging away before I mark it as reviewed.

**Pull request merged**